### ISSUE TITLE
padding-line-between-statements cannot replace lines-around-directive (no comment support)

### ISSUE THREAD

**Issue by minj** [eslint/eslint#9879]
**Tell us about your environment**

* **ESLint Version:** v4.15.0
* **Node Version:** v9.4.0
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

test.js
```js
'use strict';
/**
* comment
*/

console.log('hi');
```
test2.js
```js
'use strict';
console.log('hi');
```
test3.js
```js
// comment
'use strict';
console.log('hi');
```

```bash
eslint --no-eslintrc --rule 'padding-line-between-statements:
                             [error,
                                     {blankLine: always, prev: *, next: directive},
                                     {blankLine: always, prev: directive, next: *},
                                     {blankLine: any, prev: directive, next: directive}
                             ]' test*
```

**What did you expect to happen?**
```bash
eslint --no-eslintrc --rule 'lines-around-directive: error' test*
```
```
test2.js
  1:1  error  Expected newline after "use strict" directive  lines-around-directive

test3.js
  2:1  error  Expected newline before "use strict" directive  lines-around-directive
  2:1  error  Expected newline after "use strict" directive   lines-around-directive

test.js
  1:1  error  Expected newline after "use strict" directive  lines-around-directive
```
**What actually happened? Please include the actual, raw output from ESLint.**
test2.js
```
  2:1  error  Expected blank line before this statement  padding-line-between-statements

test3.js
  3:1  error  Expected blank line before this statement  padding-line-between-statements
```
---
It seems comments are ignored. They are not statements after all 😄 
However, without comment detection it is impossible to replace `lines-around-directive` completely.

Initially, I thought there is also a deeper problem with `prev:*, next:directive` as this failed as well:
test4.js
```js
console.log('hi');
'use strict';
console.log('hi');
```
But the syntax tree is radically different here.

This works, however:
```bash
eslint --no-eslintrc --rule 'padding-line-between-statements:
                             [error,
                                     {blankLine: always, prev: *, next: debugger}
                             ]' test5.js
```
```js
console.log('hi');
debugger;
console.log('hi');
```

**Comment by kaicataldo** [comment]
Thanks for the detailed issue

Could use the `lines-around-directive` in combination with [`lines-around-comment`](https://eslint.org/docs/rules/lines-around-comment) to achieve this?

**Comment by minj** [comment]
You misunderstood me. `lines-around-directive` works fine on its own. `lines-around-comment` works independently from it and flags the same problem twice (and I like it that way).

Seeing as `lines-around-directive` is deprecated I figured I would try to replicate it with `padding-line-between-statements`. However, this is impossible.

You could simulate it using both `padding-line-between-statements` and `lines-around-comment`, I suppose. But only if both `afterLineComment` and `beforeLineComment` are true. However, I am using a different configuration.

---

The point is, should it really be deprecated if it can't be replaced completely by a single rule?
I have now read that deprecated rules will no longer be removed only unmaintained, so at least there's that.

**Comment by kaicataldo** [comment]
Apologies, I mistyped. I meant using `padding-line-between-statements` and `lines-around-comment`, as you addressed in your reply.

It does seem like this is a case where we thought that `padding-line-between-statements` and `lines-around-comment` would be sufficient for most use cases and this particular case was missed.

As you have already pointed out, it's still possible to use deprecated rules, as we've decided to leave them in ESLint core to cause as little disruption as possible.

I'm not entirely sure this is a bug, but I'm labeling it as such for the purposes of discussion. 

@eslint/eslint-team Any other thoughts on how we should proceed here?

**Comment by Nokel81** [comment]
The following configuration is unfixable in code because of the ignoring of comments.

```json
"lines-around-comment": [1, {
    "beforeBlockComment": true,
    "afterBlockComment": false,
    "afterLineComment": false
}],
"padding-line-between-statements": [1,
    { "blankLine": "always", "prev": "*", "next": "return" },
    { "blankLine": "always", "prev": ["if", "while", "for", "try"], "next": "*" },
    { "blankLine": "never", "prev": ["let", "const"], "next": ["let", "const"] }
],
```

Code example:
```js
const x = 5;
const y = 10;

/**
 * This is a block comment
 * Still a block comment
 */
const q = "foo";
```

As it both wants and does not want the space between `y` and the `block-comment`
